api/routes/users.js

```diff
@@ -143,6 +143,27 @@ router.get(
     }
 );
 
+router.post(
+    '/profile/status',
+    async function(req, res) {
+        if (!req.body ||  !req.body.status) {
+            return res.status(400).json('Cannot update without a status update. Must also be non empty.'); 
+        }
+        if (typeof req.body.status !== 'string') {
+            return res.status(400).json('Status must be a string.'); 
+        }
```

Comment:
If i pass in white space like "        " it treats it like a valid string i think? doing a trim() check would fix this, everything else looks good to me